### PR TITLE
autogeneration into function doc

### PR DESCRIPTION
**Pull request by anoipm**
**Description**

Changes proposed in this pull request:

- Generate field description from code into CRD spec

**Related issue(s)**
See also #15002

### CODE REVIEW

File: components/function-controller/hack/function-spec-doc-gen.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+// will be replaced but rows with "<!-- KEEP-THIS -->" at the end will be preserved.
+// Special tags (html comments) must be at the end of lines due to markdown requirements.
+// All comments in markdown file that contain "<!-- SKIP-ELEMENT -->" and the name of the element
+// will skip the generation of that element. Similarly with "<!-- SKIP-WITH-ANCESTORS -->" comments,
+// which skip elements along with their descendants.
+// Words in descriptions that are surrounded by asterisks will be replaced with parameter names
+// with full path and surrounded by double asterisks.
+
+// Generated documentation is equivalent with:
+// kubectl explain --api-version='serverless.kyma-project.io/v1alpha2' function
```
Comment (anoipm) [inline components/function-controller/hack/function-spec-doc-gen.go:15]: Look on this

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 2]
```diff
@@ -111,47 +111,52 @@ spec:
 
 This table lists all the possible parameters of a given resource together with their descriptions:
 
+<!-- Part of the table bellow is generated automatically, please don't touch them. When you want to change description, do it in components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go, then regenerate using "make manifests" in components/function-controller. -->
+<!-- Some special tags (html comments) are at the end of lines due to markdown requirements. -->
+<!-- The content between "FUNCTION-SPEC-START" and "FUNCTION-SPEC-END" will be replaced but rows with "KEEP-THIS" at the end will be preserved. -->
+<!-- All comments in this file that contain SKIP-ELEMENT and the name of the element will skip the generation of that element. Similarly with SKIP-WITH-ANCESTORS comments, which skip elements along with their descendants. -->
+<!-- Words in descriptions that are surrounded by asterisks will be replaced with parameter names with full path and surrounded by double asterisks -->
```
Comment (anoipm) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:118]: Look on this

File: components/function-controller/hack/function-spec-doc-gen.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
+// This program generate part of table inside markdown file.
```
Comment (grego952) [inline components/function-controller/hack/function-spec-doc-gen.go:1]:
```suggestion
// This program generates part of the table inside the markdown file.
```

File: components/function-controller/hack/function-spec-doc-gen.go [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+// Generated table rows have format:
```
Comment (grego952) [inline components/function-controller/hack/function-spec-doc-gen.go:2]:
```suggestion
// Generated table rows have the following format:
```

File: components/function-controller/hack/function-spec-doc-gen.go [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+// | <parameter-name> | <required> | <description> |
+
+// The markdown content between html comments "<!-- FUNCTION-SPEC-START -->" and "<!-- FUNCTION-SPEC-END -->"
```
Comment (grego952) [inline components/function-controller/hack/function-spec-doc-gen.go:5]:
```suggestion
// The markdown content between HTML comments "<!-- FUNCTION-SPEC-START -->" and "<!-- FUNCTION-SPEC-END -->"
```

File: components/function-controller/hack/function-spec-doc-gen.go [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+// will be replaced but rows with "<!-- KEEP-THIS -->" at the end will be preserved.
+// Special tags (html comments) must be at the end of lines due to markdown requirements.
```
Comment (grego952) [inline components/function-controller/hack/function-spec-doc-gen.go:7]:
```suggestion
// Special tags (HTML comments) must be at the end of the lines due to the Markdown requirements.
```

File: components/function-controller/hack/function-spec-doc-gen.go [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+// All comments in markdown file that contain "<!-- SKIP-ELEMENT -->" and the name of the element
```
Comment (grego952) [inline components/function-controller/hack/function-spec-doc-gen.go:8]:
```suggestion
// All comments in the Markdown file that contain "<!-- SKIP-ELEMENT -->" and the name of the element
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 8]
```diff
@@ -111,47 +111,52 @@ spec:
 
 This table lists all the possible parameters of a given resource together with their descriptions:
 
+<!-- Part of the table bellow is generated automatically, please don't touch them. When you want to change description, do it in components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go, then regenerate using "make manifests" in components/function-controller. -->
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:114]:
```suggestion
<!-- Part of the following table is generated automatically, don't alter it. When you want to change the description, do it in components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go, then regenerate using "make manifests" in components/function-controller. -->
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 9]
```diff
@@ -111,47 +111,52 @@ spec:
...
+<!-- Some special tags (html comments) are at the end of lines due to markdown requirements. -->
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:115]:
```suggestion
<!-- Some special tags (HTML comments) are at the end of the lines due to the Markdown requirements. -->
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 10]
```diff
@@ -111,47 +111,52 @@ spec:
...
+<!-- The content between "FUNCTION-SPEC-START" and "FUNCTION-SPEC-END" will be replaced but rows with "KEEP-THIS" at the end will be preserved. -->
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:116]:
```suggestion
<!-- The content between "FUNCTION-SPEC-START" and "FUNCTION-SPEC-END" will be replaced, but the rows with "KEEP-THIS" at the end will be preserved. -->
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 11]
```diff
@@ -111,47 +111,52 @@ spec:
...
+<!-- All comments in this file that contain SKIP-ELEMENT and the name of the element will skip the generation of that element. Similarly with SKIP-WITH-ANCESTORS comments, which skip elements along with their descendants. -->
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:117]:
```suggestion
<!-- All comments in this file that contain SKIP-ELEMENT and the name of the element will skip the generation of this element. Similarly, SKIP-WITH-ANCESTORS comments skip elements along with their descendants. -->
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 2 above)
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:118]:
```suggestion
<!-- Words in the descriptions that are surrounded by asterisks will be replaced with parameter names with full path and surrounded by double asterisks -->
```

**Pull request closed without merging**